src/main/java/org/truffleruby/language/WarningNode.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2016 Oracle and/or its affiliates. All rights reserved. This
+ * code is released under a tri EPL/GPL/LGPL license. You can use it,
+ * redistribute it and/or modify it under the terms of the:
+ *
+ * Eclipse Public License version 1.0
+ * GNU General Public License version 2
+ * GNU Lesser General Public License version 2.1
+ */
+package org.truffleruby.language;
+
+import com.oracle.truffle.api.CompilerDirectives.TruffleBoundary;
+import com.oracle.truffle.api.object.DynamicObject;
+import com.oracle.truffle.api.source.SourceSection;
+import org.jcodings.specific.UTF8Encoding;
+import org.truffleruby.RubyLanguage;
+import org.truffleruby.core.rope.CodeRange;
+import org.truffleruby.core.string.StringNodes;
+import org.truffleruby.language.dispatch.CallDispatchHeadNode;
+
+/** Warns only if $VERBOSE is true. */
```

Comment:
*truthy ?

Reply:
No, `$VERBOSE` can only have 3 values, so it's only the singleton `true`:
```
[1] pry(main)> $VERBOSE=42
=> 42
[2] pry(main)> $VERBOSE
=> true
```